pkg/dns/dns.go

```diff
@@ -523,6 +523,23 @@ func getHostname(address *v1.EndpointAddress) (string, bool) {
 	return "", false
 }
 
+func (kd *KubeDNS) ensurePodSubdomain(address *v1.EndpointAddress, svc *v1.Service) bool {
```

Comment:
How frequently does this run?  I mean, it runs on every pod that needs it, but what is the period for refresh?

I'm worried about swamping the apiserver when there are lots of DNS replicas active, but we really can't watch pods.

@bowei @smarterclayton

Reply:
Does adding the subdomain to endpoints make sense for this?  (not sure whether it is general enough for endpoints having a subdomain.)